components/i2c/include/i2c.hpp

```diff
@@ -87,23 +87,26 @@ class I2c : public espp::BaseComponent {
     // Make copy of variables for easy capture
     i2c_port_t i2c_port = config_.port;
     int core_id = config_.isr_core_id;
-    if (core_id < 0) {
+    if (core_id <= 0) {
       err = i2c_driver_install(i2c_port, I2C_MODE_MASTER, 0, 0, 0);
     } else {
       if (core_id > configNUM_CORES - 1) {
         core_id = configNUM_CORES - 1;
       }
       auto isr_task = espp::Task::make_unique(espp::Task::Config{
         .name = "i2c_install",
-        .callback = [i2c_port, core_id, &err](auto &m, auto &cv) -> bool {
+        .callback = [i2c_port, core_id, this, &err](auto &m, auto &cv) -> bool {
+          std::unique_lock lock(mutex_);
           err = i2c_driver_install(i2c_port, I2C_MODE_MASTER, 0, 0, 0);
+          init_done_cv_.notify_all();
           return true; // stop the task
         },
         .stack_size_bytes = 2 * 1024,
         .priority = 15,
         .core_id = core_id,
       });
       isr_task->start();
+      init_done_cv_.wait(lock);
```

Comment:
Your mutex code is inverted here - you don't need to lock a mutex to notify.

Reply:
The intention of grabbing the mutex in the task is to synchronize the start to the init task so it doesn't begin until the first task is waiting. Although it may be possible to check the value of CV before calling wait to account for the case where the notify happened first, I feels its more tidy if the mutex ensures the intended execution order.